crates/codegen/parser/runtime/src/mod_for_destination.rs

```diff
@@ -1,15 +1,48 @@
 #[macro_use]
 pub mod parser_support;
 
-pub mod cst;
-pub mod cursor;
 pub mod kinds;
 pub mod language;
 pub(crate) mod lexer;
 pub mod parse_error;
 pub mod parse_output;
-pub mod query;
-pub mod text_index;
+pub mod queries;
 
 #[cfg(feature = "slang_napi_interfaces")]
 pub mod napi_interface;
+
+mod metaslang_cst {
+    #[derive(Clone, Debug, PartialEq, Eq, serde::Serialize)] // But why?
+    pub struct KindTypes;
```

Comment:
Whenever we want to use types as markers, it's slightly better to use an uninhabitable enum `pub enum KindTypes {}` as this makes it impossible to construct the value (by mistake), but can still be used in the generic argument position.

Reply:
done